examples/models/spatial/plot_gen_gauss.py

```diff
@@ -0,0 +1,83 @@
+r"""
+.. _generalized-gaussian-spatial-model:
+
+Generalized Gaussian Spatial Model
+==================================
+
+This is a spatial model parametrising a generalized Gaussian function.
+
+By default, the Generalized Gaussian is defined as :
+
+.. math::
+    \phi(\text{lon}, \text{lat})  = \phi(\text{r}) = N \times \exp \left[ - \left( \frac{r}{r_{\rm eff}} \right)^ \left( 1/\eta \right) \right] \,,
+
+the normalization is expressed as:
+
+.. math::
+    N = \frac{1}{ 2 \pi (1-e) r_{\rm eff}^2 \eta \Gamma(2\eta)}\,,
+where :math:`\Gamma` is the gamma function.
+This analytical norm is approximated so it may not integrate to unity in extremal cases
+if ellipticity tend to one and radius is large or :math:`\eta` much larger than one (outside the default range).
+"""
+
+# %%
+# Example plot
+# ------------
+# Here is an example plot of the model for different shape parameter:
+
+from gammapy.maps import WcsGeom, Map
+from gammapy.modeling.models import (
+    GeneralizedGaussianSpatialModel,
+    Models,
+    PowerLawSpectralModel,
+    SkyModel,
+)
+from astropy import units as u
+import matplotlib.pyplot as plt
+
+lon_0 = 20
+lat_0 = 0
+reval = 3
+dr = 0.02
+geom = WcsGeom.create(
+    skydir=(lon_0, lat_0), binsz=dr, width=(2 * reval, 2 * reval), frame="galactic",
+)
+
+tags = [r"Disk, $\eta=0.01$", r"Gaussian, $\eta=0.5$", r"Laplacian, $\eta=1$"]
+eta_range = [0.01, 0.5, 1]
+r_eff = 1
+e = 0.5
+phi = 45 * u.deg
+fig, axes = plt.subplots(1, 3, figsize=(9, 6))
+for k, eta in enumerate(eta_range):
```

Comment:
You can directly iterate over the axes here as well like `for ax, eta in zip(axes, eta_range):`